### PR TITLE
changed crate to super to fix compile error in the eg.

### PR DESCRIPTION
**Pull request by prejrdev**

### CODE REVIEW

File: src/visibility-and-privacy.md [hunk 1]
```diff
@@ -134,8 +141,7 @@ pub mod submodule {
         }
     }
 }
-
-# fn main() {}
+fn main(){};
```
Comment (ehuss) [inline src/visibility-and-privacy.md:144]: These changes should be undone, it breaks the tests.

File: src/visibility-and-privacy.md [hunk 2]
```diff
@@ -107,15 +107,22 @@ mod crate_helper_module {
...
+    use super::crate_helper_module;
+    use crate::rate_limit;
+    use super::public_api;
+    
     pub fn my_method() {
         // Any item in the local crate may invoke the helper module's public
         // interface through a combination of the two rules above.
         crate_helper_module::crate_helper();
+        public_api();
+        let doube_limit = rate_limit * 2;
```
Comment (ehuss) [inline src/visibility-and-privacy.md:125]: I'm not sure this really belongs here.  This example (especially this `my_method`) is here to illustrate the combination of the two rules above.  I'm not sure how this relates to those rules?

Comment (prejrdev) [reply src/visibility-and-privacy.md:125]: rate_limit is private only to the file itself and can't be accessed outside of it.  i added it as an example of that fact to contrast with the note about public_api()

File: src/visibility-and-privacy.md (hunk 1 above)
Comment (prejrdev) [reply src/visibility-and-privacy.md:144]: i only added that because the previous version felt like the hashtag was made as a comment and the next example had it.

Comment (prejrdev) [reply src/visibility-and-privacy.md]: ok, changed and now it passes the tests.

File: src/visibility-and-privacy.md [hunk 3]
```diff
@@ -89,9 +89,6 @@ Here's an example of a program which exemplifies the three cases outlined
 above:
 
 ```rust
-// This module is private, meaning that no external crate can access this
```
Comment (ehuss) [inline src/visibility-and-privacy.md:92]: Why was this removed?

File: src/visibility-and-privacy.md [hunk 4]
```diff
@@ -134,7 +137,6 @@ pub mod submodule {
         }
     }
 }
-
```
Comment (ehuss) [inline src/visibility-and-privacy.md:137]: This whitespace change seems unnecessary.  It would probably be best to keep consistency with the rest of the file.

File: src/visibility-and-privacy.md [hunk 5]
```diff
@@ -105,17 +102,23 @@ mod crate_helper_module {
 
 // This function is "public to the root" meaning that it's available to external
 // crates linking against this one.
-pub fn public_api() {}
+pub fn public_api(rate_limit:i32) {}
```
Comment (ehuss) [inline src/visibility-and-privacy.md:105]: It's probably best to use standard formatting.

```suggestion
pub fn public_api(rate_limit: i32) {}
```

File: src/visibility-and-privacy.md [hunk 6]
```diff
@@ -105,17 +102,23 @@ mod crate_helper_module {
...
+
+//This is by default private, and it isn't available to external crates.
```
Comment (ehuss) [inline src/visibility-and-privacy.md:107]:
```suggestion
// This is by default private, and it isn't available to external crates.
```

File: src/visibility-and-privacy.md [hunk 7]
```diff
@@ -105,17 +102,23 @@ mod crate_helper_module {
...
-
+    use super::crate_helper_module;
+    use super::MAX_RATE;
+    use super::public_api;
+    
     pub fn my_method() {
         // Any item in the local crate may invoke the helper module's public
         // interface through a combination of the two rules above.
         crate_helper_module::crate_helper();
+        public_api(MAX_RATE);
```
Comment (ehuss) [inline src/visibility-and-privacy.md:121]: It would probably be best to have a comment indicating what this is illustrating.

File: src/visibility-and-privacy.md [hunk 8]
```diff
@@ -105,17 +102,23 @@ mod crate_helper_module {
 
 // This function is "public to the root" meaning that it's available to external
 // crates linking against this one.
-pub fn public_api() {}
+pub fn public_api(rate_limit:i32) {}
+
+//This is by default private, and it isn't available to external crates.
+const MAX_RATE:i32 = 32;
```
Comment (ehuss) [inline src/visibility-and-privacy.md:108]:
```suggestion
const MAX_RATE: i32 = 32;
```

**Comment by ehuss** [comment]
I'm going to close since this is a couple years old.  Feel free to reopen if you want to address the above issues.
